src/tests/unittests/configuration_helper/adapters/test_d5a_instrument_adapter.py

```diff
@@ -61,20 +63,43 @@ def test_apply_config(self):
             self.assertEqual(d5a_module_mock(), d5a_adapter.instrument.d5a)
             self.assertEqual(instrument_name, d5a_adapter.instrument.name)
 
+            mocked_snapshot = {'name': 'd5a', 'parameters': self.mock_config}
+            d5a_adapter.instrument.snapshot = MagicMock(return_value=mocked_snapshot)
             d5a_adapter.apply(self.mock_config)
             logger_mock.assert_not_called()
-            d5a_adapter.instrument.d5a.set_voltage.assert_called_with(0, dac_value)
-            d5a_adapter.instrument.d5a.change_span_update.assert_called_with(0, 2)
+            d5a_adapter.instrument.d5a.set_voltage.assert_not_called()
+            d5a_adapter.instrument.d5a.change_span_update.assert_not_called()
             d5a_adapter.instrument.d5a.get_stepsize.assert_not_called()
             self.assertEqual(d5a_adapter.instrument.parameters['dac1'].step, 20)
             self.assertEqual(d5a_adapter.instrument.parameters['dac1'].inter_delay, 0.05)
             self.assertEqual(d5a_adapter.instrument.parameters['dac1'].unit, 'V')
 
-            self.mock_config['dac1']['value'] = None
-            d5a_adapter.apply(self.mock_config)
-            warning_text = 'Some parameter values of d5a are None and will not be set!'
-            logger_mock.warning.assert_called_once_with(warning_text)
+            d5a_adapter.instrument.close()
 
+    def test_apply_config_raises_configuration_mismatch_error(self):
```

Comment:
This is just a copy paste from the previous test right? Maybe all the duplicate code can be extracted

Reply:
Removed the extra bits